template/src/router.ts

```diff
@@ -2,45 +2,48 @@ import Vue from 'vue';
-const listComponent = () => import('./components/list').then(({ ListComponent }) => ListComponent);
-// const homeComponent = () => import(/* webpackChunkName: 'home' */'./components/home').then(({ HomeComponent }) => HomeComponent);
-// const aboutComponent = () => import(/* webpackChunkName: 'about' */'./components/about').then(({ AboutComponent }) => AboutComponent);
-// const listComponent = () => import(/* webpackChunkName: 'list' */'./components/list').then(({ ListComponent }) => ListComponent);
-
-if (process.env.ENV === 'development' && module.hot) {
-  const homeModuleId = './components/home';
-  const aboutModuleId = './components/about';
-  const listModuleId = './components/list';
-
-  // first arguments for `module.hot.accept` and `require` methods have to be static strings
-  // see https://github.com/webpack/webpack/issues/5668
-  makeHot(homeModuleId, homeComponent,
-    module.hot.accept('./components/home', () => reload(homeModuleId, (<any>require('./components/home')).HomeComponent)));
-
-  makeHot(aboutModuleId, aboutComponent,
-    module.hot.accept('./components/about', () => reload(aboutModuleId, (<any>require('./components/about')).AboutComponent)));
-
-  makeHot(listModuleId, listComponent,
-    module.hot.accept('./components/list', () => reload(listModuleId, (<any>require('./components/list')).ListComponent)));
+interface IRouteComponentConfiguration {
+  componentName: string;
+  componentPath: string;
+  urlPath: string;
 }
 
-Vue.use(VueRouter);
-
-export const createRoutes: () => RouteConfig[] = () => [
+const routeComponentConfigurationList: IRouteComponentConfiguration[] = [
   {
-    path: '/',
-    component: homeComponent,
+    'componentName': 'HomeComponent',
+    'componentPath': './components/home',
+    'urlPath': '/'
   },
   {
-    path: '/about',
-    component: aboutComponent,
+    'componentName': 'AboutComponent',
+    'componentPath': './components/about',
+    'urlPath': '/about'
   },
   {
-    path: '/list',
-    component: listComponent,
-  }
+    'componentName': 'ListComponent',
+    'componentPath': './components/list',
+    'urlPath': '/list'
+  },
 ];
 
+let routeComponentList = [];
+
+routeComponentConfigurationList.forEach((routeComponentConfiguration: IRouteComponentConfiguration) => {
+  // add empty string is necessary to get import to work
+  let routeComponent = {
+    'component': () => import('' + routeComponentConfiguration.componentPath).then((component) => component[routeComponentConfiguration.componentName]),
+    'path': routeComponentConfiguration.urlPath
+  };
+  
+  routeComponentList.push(routeComponent);
+  if (process.env.ENV === 'development' && module.hot) {
+    makeHot(routeComponentConfiguration.componentPath, routeComponent.component,
+      module.hot.accept(routeComponentConfiguration.componentPath, () => reload(routeComponentConfiguration.componentPath, (<any>require(routeComponentConfiguration.componentPath))[routeComponentConfiguration.componentName])));
```

Comment:
Unfortunately, this doesn't work for hot-reloading and fails to patch the existing component without a full page reload. 

First arguments for `module.hot.accept` and `require` methods have to be static strings. See https://github.com/webpack/webpack/issues/5668